### PR TITLE
Canard: Remove reference to HEADER_TEXT

### LINKED ISSUE #2116: Canard: Stop checking for HEADER_TEXTCOLOR in in/custom-header.php
This fails the ThemeCheck plugin's check, and prevents the theme from being updated on WordPress.org. 

```
REQUIRED: HEADER_TEXTCOLOR was found in the file custom-header.php. Use add_theme_support( 'custom-header' ) instead.
Line 37: // get_header_textcolor() options: HEADER_TEXTCOLOR is default, hide text (returns 'blank') or any hex value
Line 38: if ( HEADER_TEXTCOLOR == $header_text_color ) {
```

https://github.com/Automattic/themes/blob/773d1d70623ee503a677a1e18ab0315d22132a26/canard/inc/custom-header.php#L36-L40

The recommendation there doesn't look right, but we need to stop checking for `HEADER_TEXTCOLOR` there and find another route for verifying if that no custom header text options are set if we want to pass this check.

### PR DESCRIPTION
**Pull request by jffng**
This PR addresses #2116. There should be no visual changes.

**How to Test**

- Install Canard.
- Change the header color and verify that it works.
- Install ThemeCheck plugin.
- Run the check and verify that there are no errors.

Tested on:
WordPress 5.4.2
PHP 7.3.5

### CODE REVIEW

File: canard/inc/custom-header.php [hunk 1]
```diff
@@ -33,9 +33,10 @@ function canard_custom_header_setup() {
 function canard_header_style() {
 	$header_text_color = get_header_textcolor();
 
+	preg_match( '/^(?:[0-9a-fA-F]{3}){1,2}$/', $header_text_color, $hex_match );
```
Comment (kjellr) [inline canard/inc/custom-header.php:36]: Would you mind adding a comment here to say what this does?

**Review by kjellr — APPROVED** [review]
This seems to work fine on my end. Just left a very minor comment. @allancole would you mind taking a quick look too just to be sure? 

The general concept of this is that if the only other potential returned values of `get_header_textcolor()` are a hex color or `blank`, we can flip this conditional so that it checks against those and not against `HEADER_TEXTCOLOR`.

**Comment by allancole** [comment]
The regex might work but this solution looks a little cleaner and should still pass the checks,  @kjellr @jffng :-)  

https://github.com/Automattic/themes/blob/465f1d0de2f07eeb3d82f365ad372fe5bbd3f201/dara/inc/custom-header.php#L30-L39

**Comment by kjellr** [comment]
I updated to use the same method as [the example](https://github.com/Automattic/themes/pull/2117#issuecomment-642785987) that @allancole shared. 

Would you two mind taking another look?

**Review by allancole — APPROVED** [review]
Yup, that works for me! 

LGTM!

**Pull request merged**